pkg/automation/automation.go

```diff
@@ -72,43 +94,60 @@ func AutomateInstall() error {
 			// this is going to create a new app on each start now!
 		}
 
-		err = deleteAirgapData(clientset, appSlug)
+		license, ok := licenseSecret.Data["license"]
+		if !ok {
+			logger.Error(fmt.Errorf("license secret %q does not contain a license field", licenseSecret.Name))
+		}
+
+		decodedLicense, err := kotsutil.LoadLicenseFromBytes(license)
 		if err != nil {
-			logger.Error(errors.Wrap(err, "failed to delete airgap data"))
+			logger.Error(errors.Wrap(err, "failed to unmarshal license data"))
+		}
+
+		if decodedLicense != nil {
+			err = deleteAirgapData(clientset, decodedLicense.Spec.AppSlug)
+			if err != nil {
+				logger.Error(errors.Wrap(err, "failed to delete airgap data"))
+			}
 		}
 	}
 
 	for _, licenseSecret := range licenseSecrets.Items {
-		appSlug, err := installLicenseSecret(clientset, licenseSecret)
+		err := installLicenseSecret(clientset, licenseSecret, opts.AdditionalFiles)
 		if err != nil {
-			logger.Error(errors.Wrapf(err, "failed to install license for app %s", appSlug))
+			logger.Error(errors.Wrapf(err, "failed to install license %s", licenseSecret.Name))
 		}
-		cleanup(&licenseSecret, appSlug)
+		cleanup(&licenseSecret)
 	}
 
 	return nil
 }
 
-func installLicenseSecret(clientset *kubernetes.Clientset, licenseSecret corev1.Secret) (appSlug string, finalError error) {
+func installLicenseSecret(clientset *kubernetes.Clientset, licenseSecret corev1.Secret, additionalFiles map[string][]byte) (finalError error) {
 	license, ok := licenseSecret.Data["license"]
 	if !ok {
-		return appSlug, fmt.Errorf("license secret %q does not contain a license field", licenseSecret.Name)
+		return fmt.Errorf("license secret %q does not contain a license field", licenseSecret.Name)
 	}
 
 	unverifiedLicense, err := kotsutil.LoadLicenseFromBytes(license)
+	appSlug := ""
 	if err != nil {
 		if licenseSecret.Labels != nil {
 			appSlug = licenseSecret.Labels["kots.io/app"]
```

Comment:
nit pick, this is not needed any more, nor is the `appSlug := ""` above since we're not actually returning the slug